### PR TITLE
octave 4.4.0

### PR DESCRIPTION
**Pull request by ilovezfs**
move autoconf and automake build-time dependencies to head only
remove Java option and always require Java at build time and test time

### CODE REVIEW

**Comment by fxcoudert** [comment]
`dynare` needs revision bump. Any other reason not to merge?

File: Formula/octave.rb [hunk 1]
```diff
@@ -14,17 +13,18 @@ class Octave < Formula
...
     depends_on "icoutils" => :build
     depends_on "librsvg" => :build
     depends_on "sundials"
   end
 
   # Complete list of dependencies at https://wiki.octave.org/Building
-  depends_on "automake" => :build
-  depends_on "autoconf" => :build
   depends_on "gnu-sed" => :build # https://lists.gnu.org/archive/html/octave-maintainers/2016-09/msg00193.html
+  depends_on :java => ["1.6+", :build, :test]
```
Comment (fxcoudert) [inline Formula/octave.rb]: Unsure about that, because it means that users with non-default homebrew path now require Java to install octave.

Comment (ilovezfs) [reply Formula/octave.rb]: that seems fine to me. if that bothers them they can use the default prefix

Comment (fxcoudert) [reply Formula/octave.rb]: If they have admin rights, yes.

**Comment by ilovezfs** [comment]
@houtanb octave/config.h is now octave/octave-config.h as of Octave 4.4 so Dynare will need fixing

**Pull request closed without merging**